packages/runtime-core/src/debug.ts

```diff
@@ -0,0 +1,26 @@
+import {reactive} from '@vue/reactivity'
+import {getCurrentInstance} from './component'
+
+/**
+ * this debug function is a helper for watching states in the vue devtool (it runs only in dev mode)
+ * @example
+ * const Component = defineComponent({
+ *   setup() {
+ *  const name = ref('foo')
+ *     debug({
+ *       // watch states in the vue devtool
+ *       name,
+ *     })
+ *     return h('div', name.value)
+ *   },
+ * })
+ * @param states any states you want to see in the vue devtool
+ */
+export const debug = __DEV__ ? (states: Record<string, any>) => {
+  const instance = getCurrentInstance()
+  if (instance) {
+    instance.setupState = reactive(Object.assign({}, states, instance.setupState))
```

Comment:
What if you call it multiple times?

Reply:
I am going to make a test case for that. 👍